### PR TITLE
support data cache for webdataset

### PR DESCRIPTION
**Pull request by chenyang23333**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [x] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 1]
```diff
@@ -90,6 +90,7 @@ def __init__(
         prompt_empty_probability=0.0,
         lpw=False,
         max_embeddings_multiples=4,
+        return_sample_name=True,
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:93]: 默认值改成False

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 2]
```diff
@@ -143,11 +146,12 @@ def __init__(
                 bs_trans = instantiate_from_config(bs_trans_config)
                 self.batched_transforms.append(bs_trans)
                 print(
-                    f"Adding batch mapper {bs_trans.__class__.__name__} as batch transform #{i} " f"to the datapipeline"
+                    f"Adding batch mapper {bs_trans.__class__.__name__} as batch tran"
+                    f"sform #{i} "
+                    f"to the datapipeline"
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:151]: 打印格式改一下

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 3]
```diff
@@ -341,7 +351,7 @@ def __init__(self, shardlist_desc=None, *args, **kwargs):
         for raw in self.wds_iterator:
             try:
                 image, caption = self.parse_raw_data(raw)
-                sample = self.preprocess(image, caption)
+                sample = self.preprocess(image, caption, str(raw["__key__"]))
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:354]: 1、增加异常判断
2、用getattr获取属性值

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 4]
```diff
@@ -374,7 +384,7 @@ def __iter__(self):
...
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:387]: 同上

File: examples/stable_diffusion_xl/gm/data/dataset_wds.py [hunk 5]
```diff
@@ -397,6 +407,8 @@ class T2I_Webdataset_RndAcs(T2I_BaseDataset):
     def __init__(self, shardlist_desc=None, cache_dir=None, *args, **kwargs):
         # shardlist_desc: path to a json file describing sample num for each tar
         super().__init__(*args, **kwargs)
+        self.data_path = kwargs.get("data_path")
+        self.return_sample_name = kwargs.get("return_sample_name")
```
Comment (zhanghuiyao) [inline examples/stable_diffusion_xl/gm/data/dataset_wds.py:411]: 这两个参数在基类里面加一下

**Review by SamitHuang — APPROVED** [review]

**Review by geniuspatrick — APPROVED** [review]

**Pull request merged**